client/state/activity-log/log/reducer.js

```diff
@@ -27,3 +28,13 @@ export const logItem = ( state = undefined, { type, data, found, query } ) => {
 
 export const logItems = keyedReducer( 'siteId', logItem, [ DESERIALIZE, SERIALIZE ] );
 logItems.hasCustomPersistence = true;
+
+export const oldestItemTs = keyedReducer( 'siteId', ( state = 0, action ) => {
+	switch ( action.type ) {
+		case ACTIVITY_LOG_UPDATE:
+			return 0 < action.oldestItemTs ? action.oldestItemTs : state;
```

Comment:
I'm trying to grasp what our goal of this statement is. Although negative numbers are actually valid `Date`s we won't be having any here because they would be before 1970. Are we getting invalid timestamps somehow from the server?

If so I'd rather see us prune those in the data layer when they come in either by setting to `null` or by removing the property.

That would leave us here with the ability to use our built-in methods:

```js
return Math.min( action.oldestItemTs, state );
```

But this isn't a reason to hold up the PR. If you are interested in exploring that I'll jump right back to review when you are done.

Reply:
This was a patch for the case when `oldestItemTs` returns as `null`, which it does, after it returns a proper timestamp. It will only return the timestamp once from the server. Subsequent requests for AL data are done by the live update manager and they have the `oldestItemTs` set to `null`. So when it's `null`, I got it as `0` from the api data. In this place, if it's `0`, I'll return the previous state, which is the right timestamp we previously had.

Reply:
a null check would be great here!

```js
export const oldestItemTs = keyedReducer( 'siteId', ( state = 0, { type, oldestItemTs ) =>
	ACTIVITY_LOG_UPDATE === type && null !== oldestItemTS
		? Math.min( oldestItemTs, state )
		: state;
```